config.sh

```diff
@@ -250,78 +253,51 @@ function build_gdal {
-	        --with-hide-internal-symbols \
-	        --with-webp=${BUILD_PREFIX} \
-            --disable-debug \
-            --disable-static \
-	        --disable-driver-elastic \
-            --prefix=$BUILD_PREFIX \
-            --with-curl=curl-config \
-            --with-expat=${EXPAT_PREFIX} \
-            ${GEOS_CONFIG} \
-            --with-geotiff=internal \
-            --with-gif \
-            --with-grib \
-            --with-jpeg \
-            --with-libiconv-prefix=/usr \
-            --with-libjson-c=${BUILD_PREFIX} \
-            --with-libtiff=${BUILD_PREFIX} \
-            --with-libz=/usr \
-            --with-netcdf=${BUILD_PREFIX} \
-            --with-openjpeg \
-            --with-pam \
-            --with-png \
-            --with-proj=${BUILD_PREFIX} \
-            --with-sqlite3=${BUILD_PREFIX} \
-            --with-zstd=${BUILD_PREFIX} \
-            --with-threads \
-            --without-bsb \
-            --without-cfitsio \
-            --without-dwgdirect \
-            --without-ecw \
-            --without-fme \
-            --without-freexl \
-            --without-gnm \
-            --without-grass \
-            --without-ingres \
-            --without-jasper \
-            --without-jp2mrsid \
-            --without-jpeg12 \
-            --without-kakadu \
-            --without-libgrass \
-            --without-libgrass \
-            --without-libkml \
-            --without-mrf \
-            --without-mrsid \
-            --without-mysql \
-            --without-odbc \
-            --without-ogdi \
-            --without-pcidsk \
-            --without-pcraster \
-            --without-perl \
-            --without-pg \
-            --without-php \
-            --without-python \
-            --without-qhull \
-            --without-sde \
-            --without-sfcgal \
-            --without-xerces \
-            --without-xml2 \
-        && make -j4 \
-        && make install)
+        && mkdir build \
+        && cd build \
+        && $cmake .. \
+        -DCMAKE_INSTALL_PREFIX=$BUILD_PREFIX \
+        -DCMAKE_INCLUDE_PATH=$BUILD_PREFIX/include \
+        -DCMAKE_LIBRARY_PATH=$BUILD_PREFIX/lib \
+        -DCMAKE_PROGRAM_PATH=$BUILD_PREFIX/bin \
+        -DBUILD_SHARED_LIBS=ON \
+        -DCMAKE_BUILD_TYPE=Release \
+        -DGDAL_BUILD_OPTIONAL_DRIVERS=OFF \
+        -DOGR_BUILD_OPTIONAL_DRIVERS=OFF \
+        ${GEOS_CONFIG} \
+        -DGDAL_USE_TIFF=ON \
+        -DGDAL_USE_TIFF_INTERNAL=OFF \
+        -DGDAL_USE_GEOTIFF=ON \
+        -DGDAL_USE_GEOTIFF_INTERNAL=ON \
+        -DGDAL_ENABLE_DRIVER_GIF=ON \
+        -DGDAL_ENABLE_DRIVER_GRIB=ON \
+        -DGDAL_ENABLE_DRIVER_JPEG=ON \
+        -DGDAL_USE_ICONV=ON \
+        -DGDAL_USE_JSONC=ON \
+        -DGDAL_USE_JSONC_INTERNAL=OFF \
+        -DGDAL_USE_ZLIB=ON \
+        -DGDAL_USE_ZLIB_INTERNAL=OFF \
+        -DGDAL_ENABLE_DRIVER_NETCDF=ON \
+        -DGDAL_ENABLE_DRIVER_OPENJPEG=ON \
+        -DGDAL_ENABLE_DRIVER_PNG=ON \
```

Comment:
Perhaps enable GeoPackage (was likely enabled in autoconf build) (note: this is called OGR_ENABLE_DRIVER_GPKG but this enables both vector & raster sides)
```suggestion
        -DGDAL_ENABLE_DRIVER_PNG=ON \
        -DOGR_ENABLE_DRIVER_GPKG=ON \
```